opus/application/static_media/js/browse.js

```diff
@@ -675,56 +674,86 @@ var o_browse = {
-        let topBoxBoundary; // assign value in the each loop below to avoid getting type error in views other than #browse and #cart
+        let selector = ((opus.prefs[browse] === "gallery") ?
+                        `${tab} .gallery .thumbnail-container` :
+                        `${tab} .op-data-table tbody tr`);
+        let topBoxBoundary;
         let startObsLabel = o_browse.getStartObsLabel();
 
-        $(selector).each(function(index, elem) {
+        if ($(selector).length > 0) {
+            let namespace = opus.getViewNamespace();
+            let galleryBoundingRect = namespace.galleryBoundingRect;
+
+            // this will get the top left obsNum for gallery view or the top obsNum for table view
+            let firstCachedObs = $(selector).first().data("obs");
+            let firstCachedObsTop = $(selector).first().offset().top;
+            let calculatedFirstObs = (Math.floor((firstCachedObs - 1)/galleryBoundingRect.x + 0.0000001) *
+                                      galleryBoundingRect.x + 1);
             // Fot gallery view, the topBoxBoundary is the top of .gallery-contents
-            // For table view, we will set the topBoxBoundary to be the bottom of thead (account for height of thead)
+            // For table view, we will set the topBoxBoundary to be the bottom of thead
+            // (account for height of thead)
             topBoxBoundary = (topBoxBoundary ||
-                             (opus.prefs[browse]  === "gallery") ?
-                                $(`${tab} .gallery-contents`).offset().top :
-                                $(`${tab} .gallery-contents`).offset().top + $(`${tab} .op-data-table thead th`).outerHeight());
-
-            // compare the image .top + half its height in order to make sure we account for partial images
-            let topBox = $(elem).offset().top + $(elem).height()/2;
-            if (topBox >= topBoxBoundary) {
-                let obsNum = $(elem).data("obs");
-
-                // update obsNum in both infiniteScroll instances
-                // store the most top left obsNum in gallery for both infiniteScroll instances, this will be used to updated slider obsNum
-                $(`${tab} .op-gallery-view`).infiniteScroll({"obsNum": obsNum});
-                if (contentsView === ".op-data-table-view") {
-                    obsNum = Math.floor((obsNum-1)/o_browse.gallerySliderStep+0.0000001)*o_browse.gallerySliderStep+1;
+                              (opus.prefs[browse]  === "gallery") ?
+                              $(`${tab} .gallery-contents`).offset().top :
+                              $(`${tab} .gallery-contents`).offset().top + $(`${tab} .op-data-table thead th`).outerHeight());
+
+            // table: obsNum = calculatedFirstObs + number of row
+            // gallery: obsNum = calculatedFirstObs + number of row * number of obs in a row
+            let obsNumDiff = ((opus.prefs[browse] === "dataTable") ?
```

Comment:
Actually, no. The slug value in the URL is data, but the value in opus.prefs is dataTable. This is something Debby put in a while back when she didn't realize it would cause backwards compatibility problems with the URL. We fixed it in hash.js by doing the mapping when parsing/creating the URL. I hate this. And I just noticed a bug - hash.js only does the mapping for 'browse', not 'cart_browse'. See hash.js line 147. This needs to be fixed. I would MUCH rather get rid of dataTable entirely and just use data, though.

Reply:
All dataTable related stuff is removed, now in slug, we will have data in browse or cart_browse when we are in table view.